gix-config/src/parse/nom/tests.rs

```diff
@@ -808,7 +808,8 @@ mod key_value_pair {
     fn nonascii_is_allowed_for_values_but_not_for_keys() {
         let mut node = ParseNode::SectionHeader;
         let mut vec = Default::default();
-        assert!(key_value("你好".as_bytes(), &mut node, &mut vec).is_err());
+        assert!(key_value("你好".as_bytes(), &mut node, &mut vec).is_ok());
```

Comment:
Maybe it's explained in the PR body, but I don't understand how this can now parse OK. A second argument to `assert!()` could help clarify that especially in the presence of a test that claims that nonascii keys aren't allowed.

Besides that, I have no trouble with it as this is an internal parser and as long as the rest of the machinery works with it, all should be fine.

Reply:
Before, `section` ignored the errors from `key_value_pair, so whether an error happened or not doesn't actually matter so long as the holistic behavior was correct.

Now, we ignore `config_name` errors so we can propagate `config_value` errors past `section`.

I've added a comment in the test to clarify things.

Reply:
Perfect, thanks.